corva/middleware/unpack_context.py

```diff
@@ -0,0 +1,49 @@
+import inspect
```

Comment:
I do not understand what this whole module is for

Reply:
This module is magical, and therefore needs a lot of comments to explain concepts, choices we made, and reasoning. Please include as much detail as you can to clarify.

Reply:
Added comments

Reply:
reworked whole module